### DISCUSSION TITLE
Shut down cluster

### DISCUSSION THREAD

**Discussion by tom-asmblr** [cloudnative-pg/cloudnative-pg#1281] [Q&A]
Hello, we have been recently migrating from a different PG operator to cloudnative-pg and the experience has been fantastic.

One feature I've been unable to find is the ability to place a specific cluster in 'shutdown' mode. Effectively to safely close off the database when not in use in order to save resources. Through the plugin I've found the `hibernate` command, which I think is what I'm looking for, but would it be possible to support that at the Cluster CRD? Something like `spec.hibernate: true/false`?

Thanks again for the great operator!

**Comment by gbartolini** [comment]
Hi Tom,

We had thought of using annotations for hibernation but decided to come out first with an "imperative" way of doing it.

Alternatively, if you want to shut down the database safely, you can [fence it.](https://cloudnative-pg.io/documentation/current/fencing/) That can be done using annotations. Pods will be up, but without using compute resources (definitely disk). If you want you can scale to 1 first, then fence.

**Reply by tom-asmblr** [reply]
Hi @gbartolini, thanks for this, it's very helpful!

**Comment by gbartolini** [comment]
I am closing this issue as [declarative hibernation](https://cloudnative-pg.io/documentation/current/declarative_hibernation/) is now available.
